rgdc/rgdc/utils.py

```diff
@@ -0,0 +1,47 @@
+from datetime import datetime
+from typing import Iterator
+
+import requests
+from requests import Response, Session
+
+DEFAULT_RGD_API = 'https://www.resonantgeodata.com/api'
+
+
+def pager(session: Session, url: str, **kwargs) -> Iterator[Response]:
+    """Exhaust a DRF Paginated list."""
+    while True:
+        r = session.get(url, **kwargs)
+        yield r
+
+        if 'next' in r.links:
+            url = r.links['next']['url']
+        else:
+            break
+
+
+def results(responses: Iterator[Response]) -> Iterator[dict]:
+    for response in responses:
+        response.raise_for_status()
+        for result in response.json()['results']:
+            yield result
```

Comment:
This is pretty cool. However, I don't think it makes sense to use a generator on the user side. If doing so, you have to rerun the request if you want to access a result more than once.

Or you have to store the results of the generator yourself:

```py
q = client.search(query=json.dumps(bbox), predicate='within')
persistent = [e for e in q]
```

This is a bit cumbersome when I typically might want to iterate over the query more than once

```py
for result in q:
    print(q)
    ... # do something
    
for result in q: # <-- q is empty at this point
    print(q)
    ... # do something else
```